go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -139,7 +139,9 @@ def update_files # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity
         def run_go_mod_tidy
           return unless tidy?
 
-          command = "go mod tidy -e"
+          ## Compatibility of go1.17 and older versions.
+          ## https://golang.org/doc/go1.17#:~:text=by%20other%20commands.-,The%20-compat%20flag,-allows%20that%20version
+          command = "go mod tidy -compat=1.17 -e"
```

Comment:
Do you have a use case where adding `-compat=1.17` is necessary? I think it would be better not change this and rely on the default behavior which will determine compatibility requirements based on the version set in the `go.mod` file.

When I did a test run that only updated the go version and not `go mod tidy` it seems like we get the expected results.

go 1.16 (removes new require block):
```
 => updating github.com/shogo82148/go-retry from 1.1.0 to 1.1.1

    ± go.mod
    ~~~
    14c14
    <   github.com/shogo82148/go-retry v1.1.0
    ---
    >   github.com/shogo82148/go-retry v1.1.1
    16,28d15
    < )
    < 
    < require (
    <   github.com/aws/aws-sdk-go-v2/credentials v1.3.2 // indirect
    <   github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.4.0 // indirect
    <   github.com/aws/aws-sdk-go-v2/internal/ini v1.2.0 // indirect
    <   github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.2.2 // indirect
    <   github.com/aws/aws-sdk-go-v2/service/sso v1.3.2 // indirect
    <   github.com/aws/aws-sdk-go-v2/service/sts v1.6.1 // indirect
    <   github.com/aws/smithy-go v1.7.0 // indirect
    <   github.com/jmespath/go-jmespath v0.4.0 // indirect
    <   golang.org/x/sys v0.0.0-20191026070338-33540a1f6037 // indirect
    <   golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543 // indirect
    ~~~

    ± go.sum
    ~~~
    45,46c45,46
    < github.com/shogo82148/go-retry v1.1.0 h1:eNBmDFO0ekhF7GEx8Ie5fUmoqH/O1pSIcdivdBPLMcw=
    < github.com/shogo82148/go-retry v1.1.0/go.mod h1:NCS+03WD76BaTewJNwgyj3ukqcwNTodfy1SWHmbLZaU=
    ---
    > github.com/shogo82148/go-retry v1.1.1 h1:BfUEVHTNDSjYxoRPC+c/ht5Sy6qdwl+0kFhhubeh4Fo=
    > github.com/shogo82148/go-retry v1.1.1/go.mod h1:TPSFDcc2rlx2D/yfhi8BBOlsHhVBjjJoMvxG7iFHUbI=
    ~~~
```

go 1.17 (leaves new require block):
```
=== github.com/shogo82148/go-retry (1.1.0)
 => checking for updates 9/10
 => latest available version is 1.1.1
 => latest allowed version is 1.1.1
 => requirements to unlock: own
 => requirements update strategy: 
 => updating github.com/shogo82148/go-retry from 1.1.0 to 1.1.1

    ± go.mod
    ~~~
    14c14
    <   github.com/shogo82148/go-retry v1.1.0
    ---
    >   github.com/shogo82148/go-retry v1.1.1
    ~~~

    ± go.sum
    ~~~
    45,46c45,46
    < github.com/shogo82148/go-retry v1.1.0 h1:eNBmDFO0ekhF7GEx8Ie5fUmoqH/O1pSIcdivdBPLMcw=
    < github.com/shogo82148/go-retry v1.1.0/go.mod h1:NCS+03WD76BaTewJNwgyj3ukqcwNTodfy1SWHmbLZaU=
    ---
    > github.com/shogo82148/go-retry v1.1.1 h1:BfUEVHTNDSjYxoRPC+c/ht5Sy6qdwl+0kFhhubeh4Fo=
    > github.com/shogo82148/go-retry v1.1.1/go.mod h1:TPSFDcc2rlx2D/yfhi8BBOlsHhVBjjJoMvxG7iFHUbI=
    ~~~
```

Reply:
A fare point! Tested locally with different go versions and should agree with you